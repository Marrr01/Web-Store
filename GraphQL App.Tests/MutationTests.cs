using CookieCrumble;
using Xunit;

namespace GraphQL_App.Tests
{
    public class MutationTests
    {
        [Fact]
        public async Task ChangePassword_CorrectQuery_ReturnsUpdatedUser()
        {
            await using var result = await TestServices.ExecuteRequestAsync(b => b.SetQuery(
            """
            mutation {
              changePassword(input: { login: "login 1", newPass: "newPassword", oldPass: "password" }) {
                user {
                  login
                  password
                }
              }
            }
            """));
            result.MatchSnapshot();
        }

        [Fact]
        public async Task ChangePassword_IncorrectOldPasswordQuery_ReturnsNull()
        {
            await using var result = await TestServices.ExecuteRequestAsync(b => b.SetQuery(
            """
            mutation {
              changePassword(input: { login: "login 1", newPass: "newPassword", oldPass: "incorrectPassword" }) {
                user {
                  login
                  password
                }
              }
            }
            """));
            result.MatchSnapshot();
        }
    }
}
