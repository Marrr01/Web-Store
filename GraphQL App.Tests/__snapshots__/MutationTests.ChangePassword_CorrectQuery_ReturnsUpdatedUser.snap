{
  "data": {
    "changePassword": {
      "user": {
        "login": "login 1",
        "password": "newPassword"
      }
    }
  }
}